src/modal.ts

```diff
@@ -0,0 +1,462 @@
+}
+const animateTransform = (x:string, y:string) => {
+    return {open: {from: `transform:translate(${x}, ${y});opacity:0`,
+                   to: `opacity:1;transform:translate(0px, 0px)`},
+            close: {from: `transform:translate(0px, 0px)`,
+                    to: `transform:translate(${x},${y})`}}
+}
+const animate_defs:{ [key:string]:any} = {
+    b2t: animateTransform("0px", "100%"),
+    t2b: animateTransform('0px', "-100%"),
+    r2l: animateTransform("100%", '0px'),
+    l2r: animateTransform("-100%", '0px'),
+    fade: {open: {from: "opacity:0", to: "opacity: 1"},
+           close: {from: "opacity:1", to: "opacity:0"}}
+}
+const DEFAULT_MODAL_ZINDEX = '1';
+
+@customElement('aalam-modal')
+export class AalamModal extends LitElement {
+    DEFAULT_VALUES:{ [key:string]: any } = {
+        pos:     "xs:bottom;m:top-center",
+        height:  "xs:50vh;m:auto",
+        width:   "xs:100%;m:50vw",
+        animation: "xs:b2t",
+        position_breakpoints : [{cond:"(min-width:0px) and (max-width:992px)",
+                                 val:"bottom"},
+                                {cond: "(min-width:992px)", val: "top-center"}],
+        height_breakpoints : [{cond: "(min-width:0px) and (max-width:992px)",
+                               val: "50vh"},
+                             {cond: "(min-width:992px)", val: "auto"}],
+        width_breakpoints : [{cond: "(min-width:0px) and (max-width:992px)",
+                              val: "100%"},
+                             {cond: "(min-width:992px)", val: "50vw"}],
+        animate_breakpoints : [{cond: "(min-width:0px)", val: "b2t",
+                                ul:null, ll:0}]
+    }
+    @property({type:Boolean, reflect:true, attribute:true})
+    open = false;
+
+    @property({type:Boolean, attribute:true})
+    noesc = false;
+
+    @property({type:Boolean, attribute:true})
+    nobgclose = false;
+
+    @property({type:Boolean, attribute:true})
+    stack = false;
+
+    @property({type:String, attribute:true})
+    guidesel = null;
+
+    @property({type:String, attribute:true})
+    pos = this.DEFAULT_VALUES.pos;
+
+    @property({type:String, attribute:true})
+    height = this.DEFAULT_VALUES.height;
+
+    @property({type:String, attribute:true})
+    width = this.DEFAULT_VALUES.width;
+
+    @property({type:String, attribute:true})
+    animation = this.DEFAULT_VALUES.animation;
+
+    @property({type:String, attribute:true})
+    closesel = null;
+
+    @state()
+    animationDur = .3;
+
+    @state()
+    _position_breakpoints = this.DEFAULT_VALUES.position_breakpoints;
+
+    @state()
+    _height_breakpoints = this.DEFAULT_VALUES.height_breakpoints;
+
+    @state()
+    _width_breakpoints = this.DEFAULT_VALUES.width_breakpoints;
+
+    @state()
+    _animate_breakpoints = this.DEFAULT_VALUES.animate_breakpoints;
+
+    @state()
+    _guide_value:{ [key:string]:any} = {};
+
+    @state()
+    _open = false;
+
+    @state()
+    _mouse_pos_mod = null;
```

Comment:
This can be just a private variable instead of a state variable. If we modify the state variable, that would call render unnecessarily. We can avoid that by declaring it as a private.